src/Utils.cc

```diff
@@ -252,4 +252,51 @@ void* fbgemmAlignedAlloc(
   return aligned_mem;
 }
 
+thread_type_t
+fbgemmGetThreadPartition(int m, int n, int thread_id, int num_threads) {
+  int m_num_threads;
+  assert(num_threads >= 1);
+  // Heuristic for determine the thread partitions for parallelizing across m or
+  // n dimensions.
+  // TODO: more smart ways for thread partitions considering the
+  // grain size (MR, NR) parameters
+  if (m < n && m / num_threads < 32 &&
+      n / num_threads >=
+          128) { // Only parallelize the n dimension. Note that we have the
+                 // additional overhead of allocating the duplicated packA
+                 // buffer for each thread when parallelizing this dimension.
+    m_num_threads = 1;
+  } else if (
+      num_threads >= 4 && m / num_threads < 128 &&
+      n / num_threads >= 64) { // Parallelize both m and n dimension.
+    m_num_threads = static_cast<int>(std::sqrt(num_threads));
```

Comment:
We should prefer parallelizing across m so use ceil?

Reply:
Updated.